ql/src/java/org/apache/hadoop/hive/ql/plan/ImportTableDesc.java

```diff
@@ -0,0 +1,309 @@
+    this.table = table;
+
+    switch (getTableType()) {
+      case TABLE:
+        this.createTblDesc = new CreateTableDesc(dbName,
+                table.getTableName(),
+                false, // isExternal: set to false here, can be overwritten by the IMPORT stmt
+                table.isTemporary(),
+                table.getSd().getCols(),
+                table.getPartitionKeys(),
+                table.getSd().getBucketCols(),
+                table.getSd().getSortCols(),
+                table.getSd().getNumBuckets(),
+                null, null, null, null, null, // these 5 delims passed as serde params
+                null, // comment passed as table params
+                table.getSd().getInputFormat(),
+                table.getSd().getOutputFormat(),
+                null, // location: set to null here, can be overwritten by the IMPORT stmt
+                table.getSd().getSerdeInfo().getSerializationLib(),
+                null, // storagehandler passed as table params
+                table.getSd().getSerdeInfo().getParameters(),
+                table.getParameters(), false,
+                (null == table.getSd().getSkewedInfo()) ? null : table.getSd().getSkewedInfo()
+                        .getSkewedColNames(),
+                (null == table.getSd().getSkewedInfo()) ? null : table.getSd().getSkewedInfo()
+                        .getSkewedColValues(), null, null);
+        this.createTblDesc.setStoredAsSubDirectories(table.getSd().isStoredAsSubDirectories());
+        break;
+      case VIEW:
+        String[] qualViewName = { dbName, table.getTableName() };
+        String dbDotView = BaseSemanticAnalyzer.getDotName(qualViewName);
+        if (table.isMaterializedView()) {
+          this.createViewDesc = new CreateViewDesc(dbDotView,
+                  table.getAllCols(),
+                  null, // comment passed as table params
+                  table.getParameters(),
+                  table.getPartColNames(),
+                  false,false,false,false,
+                  table.getSd().getInputFormat(),
+                  table.getSd().getOutputFormat(),
+                  null, // location: set to null here, can be overwritten by the IMPORT stmt
+                  table.getSd().getSerdeInfo().getSerializationLib(),
+                  null, // storagehandler passed as table params
+                  table.getSd().getSerdeInfo().getParameters());
+        } else {
+          this.createViewDesc = new CreateViewDesc(dbDotView,
+                  table.getAllCols(),
+                  null, // comment passed as table params
+                  table.getParameters(),
+                  table.getPartColNames(),
+                  false,false,false,
+                  table.getSd().getInputFormat(),
+                  table.getSd().getOutputFormat(),
+                  table.getSd().getSerdeInfo().getSerializationLib());
+        }
+
+        this.createViewDesc.setViewOriginalText(table.getViewOriginalText());
```

Comment:
If REPL LOAD is applied on a renamed database, then need to update the original and expanded text here to point to destination database name else the view will still run the query on source database.